### PR TITLE
Mask urlfetch variable on commands page

### PR DESCRIPTION
**Pull request by alazymeme**
Pull request checklist:

- [x] `CHANGELOG.md` was updated

Credit to @SoWhoYou for the help

### CODE REVIEW

File: pajbot/models/command.py [hunk 1]
```diff
@@ -440,20 +440,23 @@ def autogenerate_examples(self):
             example = CommandExample(self.id, "Default usage")
             subtype = self.action.subtype if self.action.subtype != "reply" else "say"
             example.add_chat_message("say", self.main_alias, "user")
+            cleanstring = re.sub(
+                r"\$\(urlfetch ([A-Za-z0-9\-._~:/?#\[\]@!$%&\'()*+,;=]+)\)", "(urlfetch)", self.action.response
```
Comment (RAnders00) [inline pajbot/models/command.py]: Would it be possible to import this instead of duplicating it?

Comment (alazymeme) [reply pajbot/models/command.py]: Yes this shouldn't be a problem - I'm assuming you mean import it from the linkchecker module?
I can work on it tonight

Comment (alazymeme) [reply pajbot/models/command.py]: I've gotten as far as this, but my bot produces an error 500 on the commands page. Are you able to point me in the right direction?
image

**Comment by RAnders00** [comment]
@alazymeme What‘s the error in the console?

**Comment by alazymeme** [comment]
@RAnders00 ```Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]: [2020-01-16 21:32:05,016] [DEBUG  ] Updating commands...
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]: [2020-01-16 21:32:05,047] [ERROR  ] add_handler No handler for on_message found.
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]: [2020-01-16 21:32:05,066] [ERROR  ] add_handler No handler for on_pubmsg found.
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]: [2020-01-16 21:32:05,068] [ERROR  ] add_handler No handler for on_message found.
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]: [2020-01-16 21:32:05,068] [ERROR  ] add_handler No handler for on_commit found.
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]: [2020-01-16 21:32:05,080] [ERROR  ] add_handler No handler for on_usernotice found.
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]: [2020-01-16 21:32:05,103] [ERROR  ] add_handler No handler for on_pubmsg found.
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]: [2020-01-16 21:32:05,106] [ERROR  ] add_handler No handler for on_pubmsg found.
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]: [2020-01-16 21:32:05,110] [ERROR  ] add_handler No handler for on_message found.
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]: [2020-01-16 21:32:05,139] [ERROR  ] Unhandled exception
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]: Traceback (most recent call last):
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]:   File "/opt/pajbot/venv/lib/python3.6/site-packages/flask/app.py", line 1949, in full_dispatch_request
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]:     rv = self.dispatch_request()
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]:   File "/opt/pajbot/venv/lib/python3.6/site-packages/flask/app.py", line 1935, in dispatch_request
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]:     return self.view_functions[rule.endpoint](**req.view_args)
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]:   File "./pajbot/web/routes/base/commands.py", line 17, in commands
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]:     bot_commands_list = get_commands_list()
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]:   File "./pajbot/web/routes/base/commands.py", line 9, in get_commands_list
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]:     return pajbot.web.utils.get_cached_commands()
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]:   File "./pajbot/utils/time_method.py", line 18, in wrap
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]:     ret = f(*args, **kwargs)
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]:   File "./pajbot/web/utils.py", line 126, in get_cached_commands
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]:     redis.setex(commands_key, value=json.dumps(bot_commands_list, separators=(",", ":")), time=CACHE_TIME)
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]:   File "/usr/lib/python3.6/json/__init__.py", line 238, in dumps
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]:     **kw).encode(obj)
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]:   File "/usr/lib/python3.6/json/encoder.py", line 199, in encode
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]:     chunks = self.iterencode(o, _one_shot=True)
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]:   File "/usr/lib/python3.6/json/encoder.py", line 257, in iterencode
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]:     return _iterencode(o, 0)
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]:   File "/usr/lib/python3.6/json/encoder.py", line 180, in default
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]:     o.__class__.__name__)
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]: TypeError: Object of type 'Pattern' is not JSON serializable
Jan 16 21:32:05 FeelsOkayBot uwsgi[8184]: {address space usage: 270708736 bytes/258MB} {rss usage: 78020608 bytes/74MB} [pid: 8196|app: 0|req: 60/60] 159.69.37.33 () {36 vars in 476 bytes} [Thu Jan 16 21:32:05 2020] GET /commands => generated 4895 bytes in 133 msecs (HTTP/2.0 500) 2 headers in 100 bytes (1 switches on core 0)

**Comment by RAnders00** [comment]
I made a change can you try now @alazymeme you were creating a three-element tuple instead of calling .sub() for some reason

**Comment by alazymeme** [comment]
So this works, perse, but doesn't catch if statements within the urlfetch parameter
image

**Comment by RAnders00** [comment]
I dont see what exactly you would want to do differently in that screenshot, maybe you can post what output you would expect

**Comment by alazymeme** [comment]
So this is what the command currently is: https://i.nuuls.com/T90DU.png - it includes an if parameter
This is the command page output: https://i.nuuls.com/o9scn.png
This is what I think it should be: https://i.nuuls.com/4w9QE.png

**Pull request merged**